systems/plants/@RigidBodyManipulator/collisionDetect.m

```diff
@@ -136,27 +136,27 @@
 
   if ~isempty(terrain_contact_point_struct)
     xA_new = [terrain_contact_point_struct.pts];
-    idxA_new = cell2mat(arrayfun(@(x)repmat(x.idx,1,size(x.pts,2)), ...
-                                 terrain_contact_point_struct, ...
-                                 'UniformOutput',false));
-
-%    xA_new_in_world = ...
-%      cell2mat(arrayfun(@(x)forwardKin(obj,kinsol,x.idx,x.pts), ...
-%      terrain_contact_point_struct, 'UniformOutput',false));
-
-    % same as above, but also works for TaylorVar kinsols
-    tmp = arrayfun(@(x)forwardKin(obj,kinsol,x.idx,x.pts), ...
-      terrain_contact_point_struct, 'UniformOutput',false);
-    xA_new_in_world = horzcat(tmp{:});
+    
+    numStructs = size(terrain_contact_point_struct,2);
+    k = 1;
+    for i = 1:numStructs
+        numPts = size(terrain_contact_point_struct(i).pts,2);
+        for j = 1:numPts
+            terrain_pts(:, k) = terrain_contact_point_struct(i).pts(:, j);
+            terrain_idxs(k) = terrain_contact_point_struct(i).idx;
+            k = k+1;
+        end
+    end
```

Comment:
it looks like there are only a very small number of places in the code that call getTerrainContactPoints on the manipulator (most call it on the bodies explicitly).  I think it would be probably be better to just change the output be, e.g., `[pts, body_indices] = getTerrainContactPoints(manip,..)`  , where the (optional) second argument would have the same number of columns as pts.